src/utils.ts

```diff
@@ -145,29 +146,77 @@ export const getCioClient = (apiKey?: string, cioJsClientOptions?: ConstructorCl
 
 export const getActiveSectionsWithData = (
   activeSections: UserDefinedSection[],
-  sectionResults: SectionsData
+  sectionResults: SectionsData,
+  sectionsRefs: React.MutableRefObject<React.RefObject<HTMLLIElement>[]>
 ) => {
   const activeSectionsWithData: Section[] = [];
-  activeSections?.forEach((config) => {
-    const { identifier } = config;
-    let data;
+  activeSections?.forEach((sectionConfig, index) => {
+    const { identifier } = sectionConfig;
+    let sectionData;
 
-    if (isCustomSection(config)) {
+    if (isCustomSection(sectionConfig)) {
       // Copy id from data to the top level
-      data = config.data.map((item) => ({
+      sectionData = sectionConfig.data.map((item) => ({
         ...item,
         id: item?.id || item?.data?.id,
       }));
     } else {
-      data = sectionResults[identifier];
+      sectionData = sectionResults[identifier];
     }
 
-    if (Array.isArray(data)) {
-      activeSectionsWithData.push({ ...config, data });
+    if (Array.isArray(sectionData)) {
+      const section = {
+        ...sectionConfig,
+        data: sectionData,
+      };
+
+      if (sectionConfig.type === 'recommendations') {
+        // If user provided a ref in config, use it. Otherwise, use the ref from our refs array
+        section.ref = sectionConfig.ref || sectionsRefs.current[index];
+      }
+
+      activeSectionsWithData.push(section);
     }
   });
 
   return activeSectionsWithData;
 };
 
 export const escapeRegExp = (string: string) => string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+
+export const trackRecommendationView = (
+  target: HTMLElement,
+  activeSectionsWithData: Section[],
+  cioClient: Nullable<ConstructorIOClient>
+) => {
+  if (target.dataset.cnstrcRecommendationsPodId) {
+    // Pull recommendations from activeSectionsWithData by podId surfaced on target
+    const recommendationSection = activeSectionsWithData.find(
+      (section) => section.identifier === target.dataset.cnstrcRecommendationsPodId
+    );
+    const recommendationItems = recommendationSection?.data.map((item) => ({
+      itemId: item.data?.id,
+      itemName: item.value,
+      variationId: item.data?.variation_id,
+    }));
+
+    cioClient?.tracker.trackRecommendationView({
+      podId: target.dataset.cnstrcRecommendationsPodId,
+      numResultsViewed: recommendationItems?.length || 0,
```

Comment:
@esezen and @mocca102,

please see Alex S's perspective on this topic in [this slack thread from 3 months ago](https://constructor.slack.com/archives/C05LF8MTYG6/p1695227539938669?thread_ts=1694453832.486309&cid=C05LF8MTYG6): 

TLDR:
"3. how should num_results_viewed be determined?
Ideally, it would be nice to track all items in the pod (even the ones not visible at first) & corresponding num_results_viewed. Tracking just items that are visible (& corresponding num_results_viewed) is fine as well"

1. Do y'all have reason to believe that this directive has changed? 
2. Given this information, what do you think our next steps are here?

Reply:
I don't understand why we have `num_results_viewed` along with `result_count` if this is the case. We have been sending the visible count for `num_results_viewed` in the beacons. If Alex thinks `num_results_viewed` should be all, that's perfectly fine by me but I don't understand why we have 2 params. In terms of next steps, I think we are good, the current implementation sends all so we can leave it as is

Reply:
left as is

Reply:
I think we can leave the num results viewed as is but can we still add `resultCount` and `resultId`?

Reply:
Update here:
https://constructor.slack.com/archives/C05LF8MTYG6/p1702412492166239?thread_ts=1694453832.486309&cid=C05LF8MTYG6
If we finalize this PR before we hear back, we can merge and follow up in a different story